src/qt/macdockiconhandler.mm

```diff
@@ -11,52 +11,46 @@
 
 #undef slots
 #include <Cocoa/Cocoa.h>
+#include <objc/objc.h>
+#include <objc/message.h>
 
 #if QT_VERSION < 0x050000
 extern void qt_mac_set_dock_menu(QMenu *);
 #endif
 
-@interface DockIconClickEventHandler : NSObject
-{
-    MacDockIconHandler* dockIconHandler;
-}
-
-@end
-
-@implementation DockIconClickEventHandler
+static MacDockIconHandler *s_instance = NULL;
```

Comment:
No. It's a singleton pattern and i assume the OS will care about the cleanup.
But i could try to proactive free it within the destructor of `~MacDockIconHandler`. But this might be more or less the same cleanup as the os does.